lisp/ess-s-lang.el

```diff
@@ -598,8 +598,7 @@ it with the character typed if it is an ASCII character,
 otherwise remove the assignment."
   (interactive)
   (let ((assign (car ess-assign-list))
-        (event (and (numberp last-input-event)
-                    last-input-event)))
+        (event (event-basic-type last-input-event)))
```

Comment:
I guess the check for ascii below is no longer necessary then. Just check for numberp and insert.